internal/artifacts/fetch.go

```diff
@@ -113,12 +114,21 @@ func (m *Manager) fetchImageByDigest(digestRef name.Digest, architecture Arch, i
 	// verify the image signature, we only accept properly signed images
 	logger.Debug("verifying image signature")
 
-	_, bundleVerified, err := cosign.VerifyImageSignatures(ctx, digestRef, &m.options.ImageVerifyOptions)
-	if err != nil {
+	co, verified, err := verifyImageSignatures(ctx, digestRef, &m.options.ImageVerifyOptions)
+
+	if !verified {
 		return fmt.Errorf("failed to verify image signature for %s: %w", digestRef.Name(), err)
 	}
 
-	logger.Info("image signature verified", zap.Bool("bundle_verified", bundleVerified))
+	// Determine verification method and log it
+	var vm string
+	if co.SigVerifier != nil {
+		vm = "public key"
+	} else {
+		vm = "certificate subject"
+	}
+
+	logger.Info("image signature verified", zap.String("verification_method", vm))
```

Comment:
can we keep the `bundle_verified` and return it from `verifyImageSignatures`?

Reply:
Hope i got it right this time :-)